File: app/src/main/java/assignment/kz/data/network/AppApiHelper.java
package assignment.kz.data.network;


import javax.inject.Inject;
import javax.inject.Singleton;

import assignment.kz.data.network.model.LoginResponse;
import rx.Observable;

/**
 * Created by root on 4/12/17.
 */

@Singleton
public class AppApiHelper implements ApiHelper {

    @Inject
    NetworkService networkService;

    @Inject
    public AppApiHelper() {
    }

    @Override
    public Observable<LoginResponse> loginUser(String loginData) {
        return networkService.loginUser();
    }

}

File: app/src/main/java/assignment/kz/data/network/AppApiHelper.kt
package assignment.kz.data.network


import assignment.kz.data.network.model.Response
import rx.Observable
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AppApiHelper @Inject
constructor(private val networkService: NetworkService) : ApiHelper {

    override fun searchPhotos(text: String): Observable<Response> {
        return networkService.searchPhotos(text = text)
    }

}
